v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/SpannerToCassandraSourceDbIT.java

```diff
@@ -1162,4 +1197,383 @@ private void assertStringToActualRowsInCassandraDB() throws MultipleFailureExcep
+            .to(Value.json("{\"2147483647\": \"-2147483648\"}"))
+            .set("map_bigint_column")
+            .to(Value.json("{\"9223372036854775807\": 9007199254740993}"))
+            .set("map_varint_column")
+            .to(Value.json("{\"100000000000000000000\": \"-100000000000000000000\"}"))
+            .set("map_decimal_column")
+            .to(Value.json("{\"12345.6789\": \"98765.4321\"}"))
+            .set("map_ascii_column")
+            .to(Value.json("{\"example1\": \"string1\", \"example2\": \"string2\"}"))
+            .set("map_varchar_column")
+            .to(Value.json("{\"key1\": \"value1\", \"key2\": \"value2\"}"))
+            .set("map_blob_column")
+            .to(Value.json("{\"R29vZ2xl\": \"Q29tcGFueQ==\"}"))
+            .set("map_date_column")
+            .to(Value.json("{\"2025-01-27\": \"1995-01-29\"}"))
+            .set("map_time_column")
+            .to(Value.json("{\"12:30:00\": \"02:45:00\"}"))
+            .set("map_timestamp_column")
+            .to(Value.json("{\"2025-01-01T00:00:00Z\": \"9999-12-31T23:59:59.999999Z\"}"))
+            .set("map_duration_column")
+            .to(Value.json("{\"P4DT1H\": \"P4DT1H\"}"))
+            .set("map_uuid_column")
+            .to(
+                Value.json(
+                    "{\"123e4567-e89b-12d3-a456-426614174000\": \"321e4567-e89b-12d3-a456-426614174000\"}"))
+            .set("map_timeuuid_column")
+            .to(
+                Value.json(
+                    "{\"321e4567-e89b-12d3-a456-426614174000\": \"123e4567-e89b-12d3-a456-426614174000\"}"))
+            .set("map_inet_column")
+            .to(
+                Value.json(
+                    "{\"48.49.50.51\": \"::1\",\"3031:3233:3435:3637:3839:4041:4243:4445\": \"::ffff:192.0.2.128\" }"))
+            .build();
+    spannerResourceManager.write(mutation);
+  }
+
+  private void assertCassandraBoundaryData() throws InterruptedException, MultipleFailureException {
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(
+                createConfig(jobInfo, Duration.ofMinutes(10)),
+                () -> getRowCount(BOUNDARY_CONVERSION_TABLE) == 1);
+    assertThatResult(result).meetsConditions();
+    Iterable<Row> rows;
+    try {
+      rows = cassandraResourceManager.readTable(BOUNDARY_CONVERSION_TABLE);
+    } catch (Exception e) {
+      throw new RuntimeException(
+          "Failed to read from Cassandra table: " + BOUNDARY_CONVERSION_TABLE, e);
+    }
+    assertThat(rows).hasSize(1);
+    Row row = rows.iterator().next();
+    assertAll(
+        // Basic Data Types
+        () -> assertThat(row.getString("varchar_column")).isEqualTo("SampleVarchar"),
+        () -> assertThat(row.getByte("tinyint_column")).isEqualTo(Byte.MAX_VALUE),
+        () -> assertThat(row.getByte("min_tinyint_column")).isEqualTo(Byte.MIN_VALUE),
+        () -> assertThat(row.getShort("smallint_column")).isEqualTo(Short.MAX_VALUE),
```

Comment:
instead of increasing the number of columns, insert two seperate records - one with min values and one with max as in example shared earlier